staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filters
+
+import (
+	"fmt"
+	"net/http"
+	"runtime"
+
+	// TODO: decide whether to use the existing metrics, which
+	// categorize according to mutating vs readonly, or make new
+	// metrics because this filter does not pay attention to that
+	// distinction
```

Comment:
Prima facie, I'd prefer new metrics rather than repurposing existing ones in ways not originally intended. I'll dig into the request metrics a bit more to get a better idea though. 

/sig instrumentation

Reply:
Oops, that TODO note is rather outdated.  We have definitely started on using metrics appropriate for this feature.  We certainly need those.  The KEP calls out a design for those.

Reply:
Revised the comment.